framework/src/modules/chain/submodules/blocks/process.js

```diff
@@ -427,11 +444,244 @@ class Process {
 			processTransactions: scope.modules.processTransactions,
 		};
 
+		deleteLastBlock = promisify(modules.blocks.chain.deleteLastBlock);
+
 		// Set module as loaded
 		__private.loaded = true;
 	}
+
+	/**
+	 * Wrap of fork choice rule logic so it can be added to Sequence and properly tested
+	 * @param block
+	 * @param newBlockReceivedAt - Time when the new block was received since Lisk Epoch
+	 * @return {Promise}
+	 * @private
+	 */
+	async _forkChoiceTask(block, newBlockReceivedAt) {
```

Comment:
can we change this all fork choice rules to different file, and be a pure function?

Reply:
I think we can do this when the compatibility window is over as this file currently needs logic that is shared between old fork choice rules and new ones, and it will be easier to extract as old functions will be removed.